/**
* @file:   *.proto
* @author: peng
* @brief:  这个文件是通过工具自动生成的，建议不要手动修改
*/
syntax="proto3";
option go_package="./resource;resource";
package resource;

message BuyBoxConfig{
    /** 宝箱类型BuyBoxType */
    int32 BoxType = 1;
    /** 宝箱ID */
    int32 BoxID = 2;
    /** 活动子ID */
    int32 ActivitySubID = 3;
    /** 卡池排序 */
    int32 SortID = 4;
    /** 解锁需要玩家等级 */
    int32 UnlockLv = 5;
    /** 宝箱名称 */
    string BoxName = 6;
    /** 名称颜色上 */
    string BoxNameColorTop = 7;
    /** 名称颜色下 */
    string BoxNameColorDown = 8;
    /** 名称描边颜色 */
    string BoxNameOutlineColor = 9;
    /** 宝箱描述 */
    string BoxDesc = 10;
    /** 描述字体颜色 */
    string BoxDescColor = 11;
    /** 描述字体米描边颜色 */
    string BoxDescOutlineColor = 12;
    /** 宝箱图标 */
    string BoxIcon = 13;
    /** 卡池立绘或者模型ID */
    int32 BoxPortrait = 14;
    /** 卡池背景 */
    string BoxBg = 15;
    /** 入口Icon */
    string EntranceIcon = 16;
    /** 抽卡界面类型不同界面取不同数据 */
    int32 DrawCardPanelType = 17;
    /** 单抽消耗道具ID */
    int32 CostItemID = 18;
    /** 单抽消耗道具数量 */
    int32 CostItemCnt = 19;
    /** 单抽消耗货币类型 */
    int32 CostMoneyType = 20;
    /** 单抽消耗货币数量 */
    int32 CostMoneyCnt = 21;
    /** 单抽免费刷新周期 */
    int32 FreeRefreshint32erval = 22;
    /** 首次抽卡奖励 */
    int32 FirstBooty = 23;
    /** 单抽有损奖励ID */
    int32 CostBootyID = 24;
    /** 十连消耗道具ID */
    int32 TenCostItemID = 25;
    /** 十连单抽消耗道具数量 */
    int32 TenCostItemCnt = 26;
    /** 十连消耗货币类型 */
    int32 TenCostMoneyType = 27;
    /** 十连消耗货币数量 */
    int32 TenCostMoneyCnt = 28;
    /** 十连有损奖励ID */
    int32 TenCostBootyID = 29;
    /** 十连折扣 */
    float TenDiscount = 30;
    /** 抽奖券不足时是否自动补券 */
    int32 Replenish = 31;
    /** 保底次数奖励ID */
    int32 EnsureShowBootyID = 32;
    /** 解锁心愿单次数 */
    int32 UnLockWishHeroCnt = 33;
    /** 解锁召唤祝福次数 */
    int32 UnLockWishBlessHeroCnt = 34;
    /** 祝福值 */
    int32 BlessingCnt = 35;
    /** 规则文本ID */
    int32 RuleTextID = 36;
    /** 卡池概率ID */
    int32 CardPoolID = 37;
    /** 心愿池ID */
    int32 WishPoolID = 38;
    /** UP池ID */
    int32 UPPoolID = 39;
}

message BuyBoxConfigList {
    repeated BuyBoxConfig data = 1;
}

message BuyBoxChanceConfig{
    /** 卡池概率ID */
    int32 CardPoolID = 1;
    /** 英雄星级 */
    int32 HeroStar = 2;
    /** 英雄ID */
    int32 HeroID = 3;
    /** 抽取概率 */
    float Chance = 4;
}

message BuyBoxChanceConfigList {
    repeated BuyBoxChanceConfig data = 1;
}

message BuyBoxWishConfig{
    /** 心愿池ID */
    int32 WishPoolID = 1;
    /** 英雄ID */
    int32 HeroID = 2;
    /** 核心标识 */
    int32 CoreFlag = 3;
}

message BuyBoxWishConfigList {
    repeated BuyBoxWishConfig data = 1;
}

message UpBoxConfig{
    /** UP池ID */
    int32 UPPoolID = 1;
    /** 奖励集合ID */
    int32 HeroID = 2;
    /** 英雄背景 */
    string HeroBg = 3;
    /** 英雄立绘或者模型ID */
    int32 HeroModel = 4;
    /** 英雄名字图片 */
    string HeroNameBg = 5;
    /** 英雄描述文本 */
    string DescText = 6;
    /** 英雄描述文本颜色 */
    string DescTextColor = 7;
    /** 英雄描述坐标X */
    float DescPosX = 8;
    /** 英雄描述坐标Y */
    float DescPosY = 9;
    /** 英雄描述底图 */
    string HeroDescBg = 10;
    /** 保底描述底图 */
    string EnsureBg = 11;
    /** 保底描述字体描边 */
    string EnsureDescOutlineColor = 12;
    /** 保底描述字体颜色上 */
    string EnsureDescColorTop = 13;
    /** 保底描述字体颜色下 */
    string EnsureDescColorDown = 14;
    /** 单抽按钮图标 */
    string SingleDrawBtnImg = 15;
    /** UP英雄保底次数奖励ID */
    int32 EnsureID = 16;
    /** 有损奖励ID */
    int32 CostBootyID = 17;
    /** UP英雄小保底次数奖励ID */
    int32 MinEnsureID = 18;
}

message UpBoxConfigList {
    repeated UpBoxConfig data = 1;
}

